src/app/index.tsx

```diff
@@ -69,18 +68,21 @@ export interface AppOptions {
 export default (options: AppOptions) => {
   const {initialEntries, initialState} = options;
 
-  const history = typeof window !== 'undefined' && window.history
-    ? createBrowserHistory()
-    : createMemoryHistory(initialEntries && {
-      initialEntries: initialEntries.map(matchUrl),
+  const createMemoryHistoryHelper = () => {
+    const memoryHistory = createMemoryHistory(initialEntries && {
+      initialEntries: initialEntries.map(matchPathname),
     });
 
-  if (initialEntries && initialEntries.length > 0) {
-    const entry = initialEntries[initialEntries.length - 1];
-    if (hasState(entry)) {
-      history.location.state = entry.state;
+    if (initialEntries && initialEntries[0]) {
+      memoryHistory.location.state = initialEntries[0].state;
     }
-  }
```

Comment:
So it is no longer necessary for `history` created with`createBrowserHistory`?

Reply:
there was some separate `initialEntries` logic here that only actually ever worked when memoryHistory was being used, so i combined it into a helper that creates the memory history. we still use browserHistory if the window is defined